### PR TITLE
Fix and (try) to explain shadowing Issue

### PR DESCRIPTION
**Pull request by mdarrik**
Okay. So this seemed like the easiest way for me to explain what was going on. Feel free to block this or whatever you want to do. I'm sorry it's so long :cry:

So if you clone this code and run it, you should see that the colors were working appropriately. 

What I did: 

1. ## Delete `src/gatsby-theme-blog/gatsby-plugin-theme-ui/index`
     So apparently this file doesn't actually get rolled up into gatsby-plugin-theme-ui. Which is why you found making changes to it essentially did nothing. It's apparently just a helper file, that just happens to look like it's shadowing `gatsby-plugin-theme-ui/index`. In fact, there are a few files in this directory that have similar results. Here's some explanation from my friend who literally wrote the Gatsby Theme code. 
      > Gatsby theme blog creates a new set of files that introduce a new "shadow API", so shadowing anything in that sib-dir only applies to the files in Gatsby theme blog, and that doesn't roll up into Gatsby plugin theme ui's index.js
     >tldr; gatsby-theme-blog/src/gatsby-plugin-theme-ui is justsby a really similarly named file to the actual gatsby-plugin-theme-ui file
if you want to shadow a file in gatsby-plugin-theme-ui you need to shadow it, not the gatsby-theme-blog version
     > [you can only shadow] the ones that aren't in gatsby-plugin-theme-ui
colors.js is, index, components, and one other one aren't
     If you want to know more, I can put you in contact with them. I'm also planning on writing up some form of blog post on what I learned, and maybe opening an issue on the Gatsby repo about refactoring some code to clean up this confusion. 

2. ## Delete the other files in `src/gatsby-plugin-theme-ui` 
    So `color.js` and `typography.js` are helper files made by `gatsby-theme-blog`, so shadowing them doesn't actually do anything in `src/gatsby-plugin-theme-ui`. You can add them back in if you **want** to manage your colors and typography this way, or you can just add to the `colors` and `typography` properties in `src/gatsby-plugin-theme-ui/index.js`. You could also shadow `src/gatsby-theme-blog/gatsby-plugin-theme-ui`/[colors||typography].js` but that may be more confusing than it's worth in my opinion.

3. ## Merge the base blog style into your newly shadowed theme
   So your color changes weren't being applied, because by shadowing `src/gatsby-plugin-theme-ui` you weren't adding in the `styles` and the `components` objects. These objects were what applied your colors to your headers and links. You can see how this is working in the [gatsby-theme-blog repo](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-theme-blog/src/gatsby-plugin-theme-ui). But by importing and merging in `gatsby-theme-blog/gatsby-plugin-theme-ui/index` you can see the styling working. 

4. ## Uses the `sx` prop in a test link
   This was mainly just my way of confirming that everything was working. The sx prop lets you write css on a component. Feel free to switch this back to styled components or do whatever you want. But you can see that a new color was added to theme-ui. 

5. ## Delete the remaining shadowed files in `src/gatsby-theme-blog/gatsby-plugin-theme-ui`
    So as I mentioned above. The `colors` and `typography` files are actually just helpers. They get merged into your theme-ui files already, and because of `deepmerge` you can override any values you want in `src/gatsby-plugin-theme-ui/index`. However, you can do whatever makes the most sense to you. I think you can shadow `colors`, `typography`, and `prism` without a problem. For me, I think it's easier to just overwrite styles then to remember that only **some** of the files are shadowable. But, if you wanted to make a lot of changes, or discard certain values, shadowing these files may make more sense. 

I hope this helps. Feel free to ask more questions as needed, and I will do my best to answer them, or I will direct you to someone more qualified than me (which is a lot of people tbh).

### CODE REVIEW

**Comment by yurm04** [comment]
Thanks so much for you help @mdarrik !!! Your explanation was super clear and I see now where I went wrong.  Definitely encourage you to write a blog post or create a PR to update the documentation as I'm sure I'm not the only one that's struggled on this one 🙏 🙇‍♀
